doc/developer/design/20230607_shift_status_updates.md

```diff
@@ -0,0 +1,158 @@
+# Shift Storage Status Updates
+
+- [Shift Storage Status Updates](#shift-storage-status-updates)
+  - [Associated](#associated)
+  - [Context](#context)
+  - [Goals](#goals)
+  - [Overview](#overview)
+  - [Detailed description](#detailed-description)
+    - [Testing](#testing)
+    - [Rollout strategy](#rollout-strategy)
+  - [Alternatives](#alternatives)
+  - [Open questions](#open-questions)
+
+
+## Associated
+
+- [#18795](https://github.com/MaterializeInc/materialize/issues/18795)
```

Comment:
I filed a meta tracking issue since it turns out there a fair few issues related to the source/sink status views that have piled up.

```suggestion
- [#20036](https://github.com/MaterializeInc/materialize/issues/20036)
```